### ISSUE TITLE
Improve logic behind spell checking text

### ISSUE THREAD

**Issue by neomatrix369** [neomatrix369/nlp_profiler#8]
- [X] Core issue 
We have a spell checking functionality in NLP Profiler which uses a third-party library i.e. `TextBlob`, it does a decent job although the scores returned per misspelt word would then need to be correctly amortised across the whole text.

Meaning, in a fair fashion evaluate on the whole how bad is the spelling in the text. 

At the moment it's using the below logic:
```python
def spelling_quality_score(text: str) -> float:
    if (not isinstance(text, str)) or (len(text.strip()) == 0):
        return NaN

    tokenized_text = get_tokenized_text(text)
    misspelt_words = [
        each_word for _, each_word in enumerate(tokenized_text)
        if actual_spell_check(each_word) is not None
    ]
    avg_words_per_sentence = \
        len(tokenized_text) / get_sentence_count(text)
    result = 1 - (len(misspelt_words) / avg_words_per_sentence)

    return result if result >= 0.0 else 0.0
```

_Which can be improved as there are visible chances of false positive or false negative scores._

**PS: performance of this feature is being addressed on #2, so this particular issue isn't about improving it's speed/performance. Performance issues may be addressed via other issues at a later stage. There has already been some significant performance improvements to the spell check and other aspects of NLP Profiler via #2.**

Fix to #14 impacts, this issue, will need to also be fixed together.

---

- [ ] Secondary issue 

Replace the spellchecker with the package `pyspellchecker` (on [PyPi](https://pypi.org/project/pyspellchecker/)) which appears to be closer to Peter Norvig's work.

**Comment by neomatrix369** [comment]
A simpler solution would be to revert back to the original logic:

```
score = 1 - (number_of_incorrect_words / number_of_correct_words)
```
and adjust the Words of Estimative Probability table to a stricter scoring:

```
   ["Very good", 99, 100],  
    ["Quite good", 95, 99], 
    ["Good", 90, 95],  
    ["Pretty good", 85, 90], 
    ["Bad", 60, 85],  
    ["Pretty bad", 12, 60],  
    ["Quite bad", 2, 12],  
    ["Very bad", 0, 2]  
```

We can tune this logic further with new input from users in the community. Eventually, this table could be made custom or can be passed as a parameter to assist in the scoring.

**Comment by neomatrix369** [comment]
The new logic can be found in https://github.com/neomatrix369/nlp_profiler/blob/master/nlp_profiler/spelling_quality_check.py#L59 and the changes are as per the comment https://github.com/neomatrix369/nlp_profiler/issues/8#issuecomment-704932155

**Comment by neomatrix369** [comment]
Issue is partially fixed via #16.
